### PR TITLE
Add AbstractClassAssert#isPrimitive() and AbstractClassAssert#isNotPrimitive()

### LINKED ISSUE #2717: Add AbstractClassAssert#is(Not)Primitive()
#### Feature summary

The `AbstractClassAssert` lacks with `is(Not)Primitive()`.

#### Example

```java
assertThat(int.class).isPrimitive();
assertThat(String.class).isNotPrimitive();
```

There is already workarounds for the case, I know.

```
assertThat(c.isPrimitive()).isTrue();
```

I just found a case regarding the `.as(...)`.

```java
for (Field file : findSomeSpecificFields()) {
    assertThat(field.getType().isPrimitive())
        .as("@NullableBySpecification on primitive field: %s", field)
        .isFalse();
}
```
which may be more fluent,

```java
for (Field file : findSomeSpecificFields()) {
    assertThat(field.getType())
        .as("%s annotated with @NullableBySpecification, field)
        .isNotPrimitive();
    //  '...Class#field annotated with @NullableBySpecification should not be primitive.
}
```

### PR DESCRIPTION
**Pull request by ManuelG28**
#### Check List:
* Fixes #2717
* Unit tests: YES 
* Javadoc with a code example (on API only): YES 
* PR meets the [contributing guidelines](https://github.com/assertj/assertj-core/blob/main/CONTRIBUTING.md)

### CODE REVIEW

File: assertj-core/src/main/java/org/assertj/core/api/AbstractClassAssert.java [hunk 1]
```diff
@@ -800,4 +800,59 @@ public SELF hasPackage(Package aPackage) {
...
+   * assertThat(int.class).isPrimitive();
+   * assertThat(long.class).isPrimitive();
+   * assertThat(float.class).isPrimitive();
+   * assertThat(double.class).isPrimitive();
+   * assertThat(boolean.class).isPrimitive();
+   * assertThat(chart.class).isPrimitive();
+   *
+   * // this assertion fails as objects are not primitive data types:
+   * assertThat(Object.class).isPrimitive(); ;
+   * </code></pre>
```
Comment (joel-costigliola) [inline assertj-core/src/main/java/org/assertj/core/api/AbstractClassAssert.java:820]: join lines and remove double `;`

File: assertj-core/src/main/java/org/assertj/core/api/AbstractClassAssert.java [hunk 2]
```diff
@@ -800,4 +800,59 @@ public SELF hasPackage(Package aPackage) {
...
+   *
+   * // these assertions fail:
+   * assertThat(byte.class).isNotPrimitive();
+   * assertThat(short.class).isNotPrimitive();
+   * assertThat(int.class).isNotPrimitive();
+   * assertThat(long.class).isNotPrimitive();
+   * assertThat(float.class).isNotPrimitive();
+   * assertThat(double.class).isNotPrimitive();
+   * assertThat(boolean.class).isNotPrimitive();
+   * assertThat(chart.class).isNotPrimitive();
```
Comment (joel-costigliola) [inline assertj-core/src/main/java/org/assertj/core/api/AbstractClassAssert.java:848]: you need to close the tags with `</code></pre>` in the same line.
tip: generate javadoc with `mvn javadoc:javadoc` and see if it looks good

File: assertj-core/src/test/java/org/assertj/core/api/classes/ClassAssert_isNotPrimitive_Test.java [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+import static org.mockito.Mockito.verify;
+
+import org.assertj.core.api.ClassAssert;
+import org.assertj.core.api.ClassAssertBaseTest;
+
+/**
+ *  * Tests for <code>{@link ClassAssert#isNotPrimitive()} ()} ()}</code>.
+ *  *
+ *  * @author Manuel Gutierrez
```
Comment (joel-costigliola) [inline assertj-core/src/test/java/org/assertj/core/api/classes/ClassAssert_isNotPrimitive_Test.java:11]: replace `*  * ` by `* `

File: assertj-core/src/test/java/org/assertj/core/api/classes/ClassAssert_isNotPrimitive_Test.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ */
+public class ClassAssert_isNotPrimitive_Test extends ClassAssertBaseTest {
```
Comment (joel-costigliola) [inline assertj-core/src/test/java/org/assertj/core/api/classes/ClassAssert_isNotPrimitive_Test.java:13]: no need for the class to be public

File: assertj-core/src/test/java/org/assertj/core/api/classes/ClassAssert_isPrimitive_Test.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
+package org.assertj.core.api.classes;
+
+import static org.mockito.Mockito.verify;
+
+import org.assertj.core.api.ClassAssert;
+import org.assertj.core.api.ClassAssertBaseTest;
+
+/**
+ * * Tests for <code>{@link ClassAssert#isPrimitive()} ()}</code>.
```
Comment (joel-costigliola) [inline assertj-core/src/test/java/org/assertj/core/api/classes/ClassAssert_isPrimitive_Test.java:9]: replace `*  * ` by `* `

File: assertj-core/src/test/java/org/assertj/core/internal/classes/Classes_assertIsNotPrimitive_Test.java [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+    // GIVEN
+    actual = null;
+    // WHEN
+    AssertionError assertionError = expectAssertionError(() -> assertThat(actual).isNotPrimitive());
+    // THEN
+    then(assertionError).hasMessage(actualIsNull());
+  }
+
+  @Test
+  void should_fail_if_actual_has_byte_type() {
```
Comment (joel-costigliola) [inline assertj-core/src/test/java/org/assertj/core/internal/classes/Classes_assertIsNotPrimitive_Test.java:41]: there is nothing wrong with these tests but I think it's a good fit for a parameterized test taking the list of primitive type as input, this should work: 
```java
  @ParameterizedTest
  @ValueSource(classes = { int.class, double.class }) // list all primitive classes
  void should_fail_if_actual_is_a_primitive_type(Class<?> primitiveClass) {
   //WHEN
    AssertionError assertionError = expectAssertionError(() -> assertThat(primitiveClass).isNotPrimitive());
    //THEN
    then(assertionError).hasMessage(shouldNotBePrimitive(primitiveClass).create());
  }
```
Once this is done, you can remove the other `should_fail_if_actual_has_...` tests

File: assertj-core/src/test/java/org/assertj/core/internal/classes/Classes_assertIsPrimitive_Test.java [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+/**
+ * Tests for
+ * <code>{@link org.assertj.core.internal.Classes#assertIsPrimitive(AssertionInfo, Class)}</code> .
+ *
+ * @author Manuel Gutierrez
+ */
+class Classes_assertIsPrimitive_Test extends ClassesBaseTest {
+
+  @Test
+  void should_pass_if_actual_has_byte_type() {
```
Comment (joel-costigliola) [inline assertj-core/src/test/java/org/assertj/core/internal/classes/Classes_assertIsPrimitive_Test.java:23]: use a parameterized tests similar to the one I suggested in `Classes_assertIsNotPrimitive_Test`

**Review by joel-costigliola — CHANGES_REQUESTED** [review]
Thanks @ManuelG28, first round of code review done. PR looks good, just some minor/improvements comments to address.

File: assertj-core/src/test/java/org/assertj/core/api/classes/ClassAssert_isNotPrimitive_Test.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
+package org.assertj.core.api.classes;
```
Comment (joel-costigliola) [inline assertj-core/src/test/java/org/assertj/core/api/classes/ClassAssert_isNotPrimitive_Test.java:1]: add the javadoc comment with the licence, this is why the build failed.
add it to every new files

File: assertj-core/src/main/java/org/assertj/core/error/ShouldBePrimitive.java [hunk 9]
```diff
@@ -0,0 +1,34 @@
+package org.assertj.core.error;
```
Comment (joel-costigliola) [inline assertj-core/src/main/java/org/assertj/core/error/ShouldBePrimitive.java:1]: add javadoc licence

**Comment by ManuelG28** [comment]
Hi, @joel-costigliola thanks for the information about `@ParameterizedTest`. 
I have learned something new today!

Please let me know if I need to change something else.

**Comment by joel-costigliola** [comment]
Integrated thanks @ManuelG28 !

**Pull request closed without merging**